src/Entity/ProductsAwareTrait.php

```diff
@@ -21,6 +21,14 @@ public function initializeProductsCollection(): void
         $this->products = new ArrayCollection();
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function hasProducts(): bool
+    {
+        return $this->products->count() > 0;
```

Comment:
I haven't checked, but remember to mark the products association as extra lazy so that this call won't load all products (related to this brand) into memory

Reply:
Good point. Thanks